ui/src/app/archived-workflows/components/archived-workflow-list/archived-workflow-list.tsx

```diff
@@ -181,8 +181,54 @@ export class ArchivedWorkflowList extends BasePage<RouteComponentProps<any>, Sta
         return params;
     }
 
+    private filterStateObject(state: State, whitelistedStates: string[], blacklistedStates: string[]) {
+        const filteredState: State = {} as State;
+
+        if (whitelistedStates.includes('pagination') || !blacklistedStates.includes('pagination')) {
+            filteredState.pagination = state.pagination;
+        }
+
+        if (whitelistedStates.includes('namespace') || !blacklistedStates.includes('namespace')) {
+            filteredState.namespace = state.namespace;
+        }
+
+        if (whitelistedStates.includes('name') || !blacklistedStates.includes('name')) {
+            filteredState.name = state.name;
+        }
+
+        if (whitelistedStates.includes('namePrefix') || !blacklistedStates.includes('namePrefix')) {
+            filteredState.namePrefix = state.namePrefix;
+        }
+
+        if (whitelistedStates.includes('selectedPhases') || !blacklistedStates.includes('selectedPhases')) {
+            filteredState.selectedPhases = state.selectedPhases;
+        }
+
+        if (whitelistedStates.includes('selectedLabels') || !blacklistedStates.includes('selectedLabels')) {
+            filteredState.selectedLabels = state.selectedLabels;
+        }
+
+        if (whitelistedStates.includes('minStartedAt') || !blacklistedStates.includes('minStartedAt')) {
+            filteredState.minStartedAt = state.minStartedAt;
+        }
+
+        if (whitelistedStates.includes('maxStartedAt') || !blacklistedStates.includes('maxStartedAt')) {
+            filteredState.maxStartedAt = state.maxStartedAt;
+        }
+
+        if (whitelistedStates.includes('error') || !blacklistedStates.includes('error')) {
+            filteredState.error = state.error;
+        }
+
+        if (whitelistedStates.includes('deep') || !blacklistedStates.includes('deep')) {
+            filteredState.deep = state.deep;
+        }
+
+        return filteredState;
+    }
+
     private saveHistory() {
-        this.storage.setItem('options', this.state, {} as State);
+        this.storage.setItem('options', this.filterStateObject(this.state, [], ['workflows']), {} as State);
         const newNamespace = Utils.managedNamespace ? '' : this.state.namespace;
```

Comment:
I do like the way you abstracted this into a function

Reply:
Although, I'm concerned about the way the function lists out each member individually. There seems to be some maintenance risk there if somebody adds a new field to the struct, they need to update this function, right? Is there any way to do this in a way that doesn't require the maintenance?

Reply:
Maybe we should consider simply clearing the value for `this.state.workflows` and perhaps this is a little over-designed? (I'm open, though)

Reply:
Agree regarding the maintenance risk if the state expands - I've added looping to blacklist the keys instead of statically going through each key via multiple if conditions